strings/base_coroutine_foundation.h

```diff
@@ -743,7 +743,7 @@ WINRT_EXPORT namespace winrt
 
         auto [delegate, shared] = impl::make_delegate_with_shared_state<impl::async_completed_handler_t<T>>(shared_type{});
 
-        auto completed = [&](T const& async)
+        auto completed = [delegate = std::move(delegate)](T const& async)
```

Comment:
Clang didn't like the fact that a local binding `delegate` was being referenced by the lambda, leading to a Clang error.

Reply:
Not 100% sure the lifetime here is ok given all the sharing going on. @oldnewthing to confirm.

Reply:
It's okay. The important thing is that the `delegate` lifetime covers the lifetime of `shared`. The old code managed its lifetime explicitly. Now, we move it into `completed`, so `completed` controls the lifetime. But we control the lifetime of `completed` so everything is still okay: We keep `completed` alive for as long as we need `shared`. (There is a slight issue with future maintenance window because `completed` will destruct before `shared`, so anybody who uses `shared` during that window of vulnerability will be a UAF. But that isn't happening in the code right now, so we're okay.)